src/main.c

```diff
@@ -49,6 +59,56 @@ static void softReset_button_callback(struct device *port, struct gpio_callback
 	soft_reboot();
 }
 
+void cdc_acm_descriptor_callback (cdc_acm_cfg_t *cfg)
+{
+	customer_data_t *otp_data = (customer_data_t *)CUSTOMER_DATA_ADDRESS;
+
+	memset(cfg->vendor_string, '\0', 32);
+	memset(cfg->product_string, '\0', 32);
+	memset(cfg->serial_string, '\0', 32);
+
+	//read the USB descriptors from OTP and use some default values if it has not been properly populated
+	if((otp_data->vendor_name_len>0) && (otp_data->vendor_name_len<=32))
+	{
+		memcpy(cfg->vendor_string, otp_data->vendor_name, otp_data->vendor_name_len);
+	}
+	else
+	{
+		memcpy(cfg->vendor_string, vendor, strlen(vendor));
+	}
+
+	if((otp_data->board_name_len>0) && (otp_data->board_name_len<=32))
+	{
+		memcpy(cfg->product_string, otp_data->board_name, otp_data->board_name_len);
+	}
+	else
+	{
+		memcpy(cfg->product_string, product, strlen(product));
+	}
+
+	if((otp_data->product_sn_len>0) && (otp_data->product_sn_len<=32))
```

Comment:
In fact, better yet, don't used a hardcoded value. Just use sizeof to check the size of the specific field, i.e. `sizeof(otp_data->product_sn)`